iceoryx_utils/include/iceoryx_utils/internal/concurrent/lockfree_queue/index_queue.inl

```diff
@@ -209,6 +223,60 @@ bool IndexQueue<Capacity, ValueType>::popIfFull(ValueType& index) noexcept
     return false;
 }
 
+template <uint64_t Capacity, typename ValueType>
+bool IndexQueue<Capacity, ValueType>::popIfSizeIsAtLeast(uint64_t requiredSize, ValueType& index) noexcept
+{
+    if (requiredSize == 0)
+    {
+        return false;
+    }
+
+    // which to load first is up to discussion, for correctness it should make no difference
+    // but for performance it might
+    // note that without sync mechanisms (such as seq_cst), reordering is possible
+    auto writePosition = m_writePosition.load(std::memory_order_relaxed);
+    auto readPosition = m_readPosition.load(std::memory_order_relaxed);
+
+    // note: In principle it should be possible to load this just before the value
+    // is required in a successful return.
+    // However, doing this leads to failure of some stress tests for unkwown reasons
+    // This suggests that there is an issue which is not understood and needs to be investigated.
+    auto value = loadvalueAt(readPosition, std::memory_order_relaxed);
```

Comment:
No, I explicitly decided against this for readability. This would lead to code duplication and from the naming I consider it more than sufficient to know that helper function does. I did this already in the past for more complex constructs involving compare exchange, but here I do not consider it useful.

Rejected.

Reply:
@MatthiasKillat how would it lead to code duplication if it is only one line? 
This argument is valid if you have multiple lines or a complex algorithm but with one line? 
With this argument leaving this method would also lead to code duplication since you are calling `loadvalueAt` multiple times?! ;)

The readability argument is a valid one. My counter argument would be to understand the complexity of your memory synchronization and which atomic you do not synchronize in there (with `memory_order_relaxed`) and why this is valid.

I think we need a referee which decides the outcome... 
@elBoberido what is your opinion?

Reply:
@elfenpiff I have to flip a coin. I'm also a bit torn by this. On the one hand, it makes the code a bit more readable and it's also clear that a value is loaded, maybe just not from where. Have to sleep a night about this.

Reply:
It is about intention, more than anything else. Once I understand what loadValueAt does, the code is more readable.